pkg/utils/nodeutils.go

```diff
@@ -60,13 +63,50 @@ func GetFirstNodeRole(nodeSelector map[string]string) string {
 
 // AnyMcfgPoolLabelMatches verifies if the given nodeSelector matches the nodeSelector
 // in any of the given MachineConfigPools
-func AnyMcfgPoolLabelMatches(nodeSelector map[string]string, poolList *mcfgv1.MachineConfigPoolList) bool {
+func AnyMcfgPoolLabelMatches(nodeSelector map[string]string, poolList *mcfgv1.MachineConfigPoolList) (bool, *mcfgv1.MachineConfigPool) {
+	foundPool := &mcfgv1.MachineConfigPool{}
 	for i := range poolList.Items {
 		if McfgPoolLabelMatches(nodeSelector, &poolList.Items[i]) {
-			return true
+			return true, &poolList.Items[i]
 		}
 	}
-	return false
+	return false, foundPool
+}
+
+// isMcfgPoolUsingKC check if a MachineConfig Pool is using a custom Kubelet Config
+// if any custom Kublet Config used, return name of generated latest KC machine config from the custom kubelet config
+func IsMcfgPoolUsingKC(pool *mcfgv1.MachineConfigPool) (bool, string, error) {
```

Comment:
let's add unit tests for this function

Reply:
Sounds good, I will add unit tests for those as well as the e2e tests!